### DISCUSSION TITLE
Can I use data (from the cascade) in a filter? (v1.0)

### DISCUSSION THREAD

**Discussion by mbforbes** [11ty/eleventy#2841] [Q&A] [answered]
Hi, sincere apologies as I am probably missing something really obvious. (I'm on Eleventy v1.0)

## Question
Where can I find the data from the Data Cascade when writing a filter (`eleventyConfig.addFilter("foo", (arr) => {}`)?

## Longer Question
I'm building a filter (using Nunjucks, but I figured the answer wouldn't depend on that; spoilers it might), and in order to perform the filter, I need some global data.

Specifically, I'm using a filter to sort by a frontmatter attribute, but I would like to sort things differently depending on the value of the attribute. To be concrete, some posts are part of a "series," but different "series" get sorted differently---some by date, some by title, some by a provided attribute. I'm storing information about each series separately so each post doesn't have to duplicate the same metadata.

```js
eleventyConfig.addNunjucksFilter("dateSeriesGroupBy", (arr) => {
    // sort by a frontmatter property, but use data from the cascade
    // to determine how to sort
}
```

## What I've tried so far
- `console.log(Object.getOwnPropertyNames(this));`
- `console.log(Object.getOwnPropertyNames(eleventyConfig));`
- `console.log(x)` for each `x` on `eleventyConfig`;
- `console.log(eleventyConfig.globalData)` (it's `{}`)
- read a bunch of pages on Eleventy's documentation, e.g., on data cascade, collections, filters
- searched for related GitHub Discussions

Thank you for any help!

**Comment by pdehaan** [comment, marked as answer]
Not 100% sure I understand what you're asking for, but there is https://www.11ty.dev/docs/filters/#scoped-data-in-filters which recommends `function (arr) {...}` syntax versus arrow functions `(arr) => {...}` so you have some additional variables in `this.` scope.

But let me try and build a better example in a second and get back to you with some Nunjucks and LiquidJS and .11ty.js ideas (for other languages you're on your own).

**Reply by pdehaan** [reply]
https://github.com/pdehaan/11ty-2841

When using .njk templates, I had to use `this.ctx` in my filter.
When using .liquid templates, I had to use `this.context.environments` in my filter.
When using .11ty.js templates, I had to pass the data manually to my filter. :shrug:

**Reply by mbforbes** [reply]
Thank you so much!
- Re: what I'm asking for, yes, sorry, I erred too much on the side of brevity. I will update my question for posterity. However, you still figured out what I wanted and answered it.
- I'd seen that in the documentation, but since both the properties listed said `Added in v2.0.0`, I didn't realize there were _more_ properties that existed, _and_ they also exited before v2.0.0. 
- Since I'm using Nunjucks, switching from the arrow function to plain syntax gave me access to `this.ctx`, which contained everything I needed.
- I'll open a docs bug to hopefully add this.

Agin, really appreciate you taking the time to setup a whole example for me.
